internal/gatewayapi/translator.go

```diff
@@ -546,11 +548,129 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						}
 					}
 
+					// TODO: implement header modifier filter
+					for _, filter := range rule.Filters {
+						if irRoute.DirectResponse != nil {
+							// If an invalid filter type has been configured then an error direct response must be returned
+							// so skip processing any other filters for this route
+							break
+						}
+						switch filter.Type {
+						case "RequestRedirect":
```

Comment:
nit, there are constants for the filter types defined in Gateway API:

```suggestion
						case v1beta1.HTTPRouteFilterRequestRedirect:
```